### PR TITLE
LibJS: throw RangeError in `StringPrototype::repeat` if OOM

### PR DESCRIPTION
**Pull request by ttrssreal**
Currently crashes with an assertion failure in `String::repeated` if malloc can't serve a `count * input_size` sized request, so propagate the error then thow a `RangeError`.

This can happen when
```js
"A".repeat(0xffffffffff);
```
```
VERIFICATION FAILED: !_temporary_result.is_error() at /home/jess/code/serenity/AK/String.cpp:313
/home/jess/code/serenity/Build/lagom/bin/../lib64/liblagom-ak.so.0(ak_verification_failed+0x86) [0x7ffff7767f2f]
/home/jess/code/serenity/Build/lagom/bin/../lib64/liblagom-ak.so.0 AK::String::repeated(AK::String const&, unsigned long) 0x5e) [0x7ffff77983ba]
/home/jess/code/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::StringPrototype::repeat(JS::VM&) 0x3bf) [0x7ffff7b9f3bb]
Build/lagom/bin/js() [0x40e707]
/home/jess/code/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0(+0x335291) [0x7ffff7b35291]
/home/jess/code/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::NativeFunction::call() 0x3d) [0x7ffff7b3471d]
/home/jess/code/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::NativeFunction::internal_call(JS::Value, AK::Span<JS::Value const>) 0x1be) [0x7ffff7b34c24]
/home/jess/code/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::call_impl(JS::VM&, JS::FunctionObject&, JS::Value, AK::Span<JS::Value const>) 0x30) [0x7ffff7a20827]
/home/jess/code/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0(+0x18e2d1) [0x7ffff798e2d1]
/home/jess/code/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::Bytecode::Op::Call::execute_impl(JS::Bytecode::Interpreter&) const 0x4c6) [0x7ffff7981ec8]
/home/jess/code/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::Bytecode::Interpreter::run_bytecode() 0x997) [0x7ffff7987209]
/home/jess/code/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::Bytecode::Interpreter::run_and_return_frame(JS::Bytecode::Executable&, JS::Bytecode::BasicBlock const*, JS::Bytecode::CallFrame*) 0x198) [0x7ffff7987d06]
/home/jess/code/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::Bytecode::Interpreter::run(JS::Script&, JS::GCPtr<JS::Environment>) 0x3be) [0x7ffff79884da]
...
```
Its not [spec'ed](https://tc39.es/ecma262/#sec-string.prototype.repeat), but V8 and SpiderMonkey both do somthing similar with a max string length, I think its better than crashing, maybe a fixed max string length is better?
They also both throw `RangeError`'s

### CODE REVIEW

File: AK/String.cpp [hunk 1]
```diff
@@ -310,11 +310,17 @@ bool String::equals_ignoring_ascii_case(StringView other) const
 
 String String::repeated(String const& input, size_t count)
 {
-    VERIFY(!Checked<size_t>::multiplication_would_overflow(count, input.bytes().size()));
+    return MUST(String::repeated_with_error(input, count));
+}
+
+ErrorOr<String> String::repeated_with_error(String const& input, size_t count) {
```
Comment (Hendiadyoin1) [inline AK/String.cpp:316]: We usually name something like this `try_repeated`

File: AK/String.cpp [hunk 2]
```diff
@@ -310,11 +310,17 @@ bool String::equals_ignoring_ascii_case(StringView other) const
...
+    if (Checked<size_t>::multiplication_would_overflow(count, input.bytes().size())) {
+        return Error::from_string_literal("String::repeated: count too large");
```
Comment (Hendiadyoin1) [inline AK/String.cpp:318]: Maybe `EOVERFLOW` instead

**Comment by awesomekling** [comment]
Fix seems okay, but please add a test :)

File: Userland/Libraries/LibJS/Runtime/ErrorTypes.h [hunk 3]
```diff
@@ -230,6 +230,7 @@
     M(StringNonGlobalRegExp, "RegExp argument is non-global")                                                                           \
     M(StringRawCannotConvert, "Cannot convert property 'raw' to object from {}")                                                        \
     M(StringRepeatCountMustBe, "repeat count must be a {} number")                                                                      \
+    M(StringRepeatCountTooBig, "repeat count too big")                                                                                  \
```
Comment (trflynn89) [inline Userland/Libraries/LibJS/Runtime/ErrorTypes.h:233]: These error strings are visible to the user, so let's be a bit more descriptive in the message. Something like:
```
M(StringRepeatCountMustNotOverflow, "repeat count must not overflow")
```
(which matches what other engines display on this error)

**Review by trflynn89 — APPROVED** [review]

**Comment by ttrssreal** [comment]
sweet :3

**Comment by ttrssreal** [comment]
ASAN doesn't like the big allocation?

**Pull request merged**